src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,207 @@
+package duke;
+
+import java.io.IOException;
+import java.time.LocalDate;
+import java.util.Scanner;
+
+/**
+ * <h1>duke.Duke!</h1>
+ * The duke.Duke program implements an application that
+ * helps you keep track of your tasks.
+ *
+ * @author Augustine Kau
+ * @version 0.1
+ * @since 2020-08-18
+ */
+public class Duke {
+
+    private final Storage storage;
+    private final TaskList ls;
+    private final Ui ui;
+
+    /**
+     * Command that user can input
+     */
+    enum Command {
+        BYE, LIST, DONE, TODO, DEADLINE, EVENT, DELETE, CHECK, FIND
+    }
+
+    /**
+     * Constructor for duke.Duke
+     * @param filePath For file todolist.txt.
+     * @throws IOException On input error.
+     */
+    public Duke(String filePath) throws IOException {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        ls = new TaskList(storage.loadFile());
+    }
+
+    /**
+     * This method is used to initiate the chat bot.
+     * @throws IOException On input error.
+     */
+    public void run() throws IOException {
+        ui.showWelcome();
+        // Using Scanner for Getting Input from User
+        Scanner in = new Scanner(System.in);
+
+        WhileLoop:
```

Comment:
Does this WhileLoop: do anything in the code? I have never seen it before so I am not sure